src/react-native/07-debugging-devtools/debugging-devtools.md

```diff
@@ -10,39 +10,78 @@
 
 In this section, you will:
 
-- TODO
+- [resource](https://reactnative.dev/docs/debugging)
 
-## Objective 1: TODO
+## Objective: Devtools setup
 
-TODO
+We will be setting up dev tools to help us debug and inspect the UI:
 
-### Concept TODO
+**React Native Debugger**
 
-TODO
+<img alt="Screenshot of react native debugger." src="../../static/img/react-native/07-debugging-devtools/react-native-debugger
+.png"/>
 
-### Setup 1
+**React Devtool**
 
+<img alt="Screenshot of a connected dev tool." src="../../static/img/react-native/07-debugging-devtools/dev-tool-connected.png"/>
+
+### Concept TODO
 TODO
 
-### Verify 1
+### Setup
 
-TODO
+✏️ Update **src/package.json** to be:
 
-### Exercise 1
+@diff ../../../exercises/react-native/06-custom-components/03-solution/package.json ../../../exercises/react-native/07-debugging/01-solution/package.json only
 
-TODO
+Since we have a new package let's run the following command.
 
-### Solution 1
+```shell
+npm i
+```
 
-If you’ve implemented the solution correctly, the tests will pass when you run `npm run test`!
+### Verify
 
-<details>
-<summary>Click to see the solution</summary>
+**Verifying React Native Debugger**
 
-TODO
+Execute the command:
+
+```shell
+npm run debugger
+```
+
+The application should be running in Android Studio. Open the terminal and run the following command:
+
+```shell
+adb shell input keyevent 82
+```
+
+This will open up the dev menu. It should look like the following image:
+
+<img alt="Screenshot of a connected dev tool." src="../../static/img/react-native/07-debugging-devtools/in-app-dev-menu.png" style="max-height: 750px; border: 4px solid black; border-radius: 25px;"/>
+
+Clicking on `Open Debugger` will open up a separate window for debugging.
+
+**Verifying React Devtool**
+
+While the application is running execute the command:
+
+```shell
+npm run devtools
+```
+
+This should open a separate window that looks like the following image:
+
+<img alt="Screenshot of a connected dev tool." src="../../static/img/react-native/07-debugging-devtools/dev-tool-not-connected.png"/>
```

Comment:
Screenshot of a window with options for React Native and React DOM. The React Native section suggests opening the in-app developer menu. A notification at the top states the server is waiting for React to connect.